screens/Dashboard.js

```diff
@@ -29,20 +22,41 @@ class Dashboard extends Component {
     viewGroup = (val) => {this.props.navigation.navigate('View Group',{groupID: val})}
     
     componentDidMount() {
+      this.setState({loading: true});
       this.setState({groups: []});
-      const groupArray  = [];
-      firebase.database().ref('/groups').on('value', (snapshot) => {
-        snapshot.forEach(function (childSnapshot) {
-          groupArray.push(childSnapshot.toJSON());
+      
+
+      firebase.database().ref('/groups').on('value', snapshot => {
+        const groupArray =[];
+        //console.log(snapshot.toJSON());
+        //console.log('groupArray pre childSnapshot: ',groupArray);
+        snapshot.forEach((childSnapshot) => {
+          //console.log(childSnapshot.toJSON());
+          const dict = childSnapshot.toJSON();
+          console.log('dict: ',dict);
+          if (dict.hasOwnProperty("goal") &&
+          dict.hasOwnProperty("groupColor") &&
+          dict.hasOwnProperty("groupFreq") &&
+          dict.hasOwnProperty("groupID") &&
+          dict.hasOwnProperty("groupMemberNames") &&
+          dict.hasOwnProperty("groupName") &&
+          dict.hasOwnProperty("streak") && 
```

Comment:
are these checks necessary? I thought the data from childSnapShot is guaranteed to have these fields (once we put in the verification in the createGroup page).